contracts/utils/Address.sol

```diff
@@ -7,6 +7,32 @@ pragma solidity ^0.8.19;
  * @dev Collection of functions related to the address type
  */
 library Address {
+    /**
+     * @dev The ETH balance of the account is not enough to perform the operation.
+     */
+    error AddressInsufficientBalance(address account);
+
+    /**
+     * @dev A call to `target` failed. The `target` may have reverted.
+     */
+    error AddressFailedCall(address target);
+
+    /**
+     * @dev A low level call failed without any further reason.
+     */
+    error AddressFailedLowLevelCall();
```

Comment:
I was surprised that we have a "failed call" and a "failed low level call" that are different. I don't think this difference makes sense and I would like this to be the same error. There is one issue with the `onRevert` function, it does not receive any arguments so we can't keep the `target` argument. I'm fine with not having it. (We could also add it as an argument to `onRevert`, but it complicates `verifyCallResult` because the target address is not available.)

Reply:
Correct, we had the same train of thought. My solution was having two different errors but I agree it's weird.

Would you rather have two errors or only one? If only one, is it okay if none of them have `target` or how would we add `target` to `AddressFailedLowLevelCall`?

Reply:
Actually my main issue is the "low level call" part, because they are all low level calls, and it doesn't even matter for the end user.

In order to add the `target`, we have to do this:

```solidity
    function verifyCallResultFromTarget(
        address target,
        bool success,
        bytes memory returndata,
        function(address target) internal view customRevert
    ) internal view returns (bytes memory) {
        if (success) {
            if (returndata.length == 0) {
                // only check if target is a contract if the call was successful and the return data is empty
                // otherwise we already know that it was a contract
                if (target.code.length == 0) {
                    revert AddressEmptyCode(target);
                }
            }
            return returndata;
        } else {
            _revertFromTarget(target, returndata, customRevert);
        }
    }

    function _revertFromTarget(address target, bytes memory returndata, function(address target) internal view customRevert) private view {
        // Look for revert reason and bubble it up if present
        if (returndata.length > 0) {
            // The easiest way to bubble the revert reason is using memory via assembly
            /// @solidity memory-safe-assembly
            assembly {
                let returndata_size := mload(returndata)
                revert(add(32, returndata), returndata_size)
            }
        } else {
            customRevert(target);
            revert AddressExpectedRevert();
        }
    }
```

But for `verifyCallResult` we can't do this, because we don't have a target. We could add a target to that function. `verifyCallResultFromTarget` is unfortunately named similarly even though it has a different behavior.

Alternatively, remove the target argument from the error, this is not so bad in my opinion.

Reply:
> and it doesn't even matter for the end user.

I agree with this. However, I wouldn't add the `target` parameter, correct me if wrong but that restrict users to always providing a target right? I'd rather have no argument than mandatory arguments.

> Alternatively, remove the target argument from the errors, this is not so bad in my opinion.

I partially agree with this because, from the `Address.sol` perspective, it's very clear what the target may be, but if another system is using `Address.sol` (eg. Governor), the `target` may be valuable information.

I pushed a commit merging both errors into `AddressFailedCall` because I still think that option is the best from what he have.